src/controls/utils.ts

```diff
@@ -12,6 +12,23 @@ export const coerceToInteger = (value: number) => Math.round(value)
 
 export const coerceToNumber = (value: number) => Number(value)
 
-export const hasLeadingZero = (value?: string) =>
-  // The value 0 is just 0 and doesn't have leading 0s.
-  value && value.substring(0, 1) === "0" && value.length > 1
+export const areStringNumbersEqual = (
```

Comment:
We actually don't care if the input has leading zeros. What we are actually trying to determine is if the inputs are equal when parsed to numbers.